cypress/support/commands/application.js

```diff
@@ -1,13 +1,45 @@
-Cypress.Commands.add('testResponseCodeAndType', (path, responseCode, type) => {
-  cy.request({ url: path, failOnStatusCode: false }).then(
-    ({ status, headers }) => {
-      expect(status).to.eq(responseCode);
-      expect(headers['content-type']).to.include(type);
-      // Always ensure we're not seeing the Mozart fallback
-      expect(
-        headers,
-        `Mozart fallback response detected for ${path}`,
-      ).not.to.have.property('x-mfa');
-    },
-  );
-});
+// Custom command to test that the specified path returns the expected status code and
+// content type, as well as that's the response was not a Mozart fallback. Automatically
+// retries twice after a delay if it gets an unexpected response
+// NB:
+// - Default timeout for cy.request is 50s
+// - Certain types of network error are retried automatically (retryOnNetworkFailure)
+Cypress.Commands.add(
+  'testResponseCodeAndType',
+  (path, responseCode, type, retriesLeft = 2) => {
+    cy.request({ url: path, failOnStatusCode: false }).then(
+      ({ status, headers }) => {
+        try {
+          expect(status, `Unexpected status code for ${path}`).to.equal(
+            responseCode,
+          );
+          expect(
+            headers['content-type'],
+            `Unexpected content-type for ${path}`,
+          ).to.include(type);
+          // Always ensure we're not seeing the Mozart fallback (denoted by x-mfa header)
+          expect(
+            headers,
+            `Mozart fallback response detected for ${path}`,
+          ).not.to.have.property('x-mfa');
```

Comment:
I think this approach may hide a certain class of errors that could be important. If a request to a page intermittently returns incorrectly (regardless of status code), not due a fallback response being returned, it would be ignored and retried now. Could the retry be restricted to detect the `x-mfa` header only rather the the other assertions (https://github.com/bbc/simorgh/pull/5056/files#diff-7b1cde41ca5fe3c5fe4861d1c706eb5bR13-R19)? That way,  this change would only massage flakey fallback responses but not genuine incorrect responses.

Sorry, found that rather hard to explain, give me a shout if it doesn't make sense @jamesdonoh

Reply:
Ah interesting so just so I'm clear, you are suggesting we retry for fallbacks but not the other two types of assertion failure (status and content type)? That would certainly be a more conservative approach. I'll have a go!